routing/payment_lifecycle.go

```diff
@@ -262,6 +262,16 @@ lifecycle:
 		// We found a route to try, create a new HTLC attempt to try.
 		attempt, err := p.registerAttempt(rt, ps.RemainingAmt)
 		if err != nil {
+			// If the error is due to we cannot register another
```

Comment:
I think we should in neither case when receiving an error also in requestRoute just exist the payment without marking it as failed not collecting all results. So what we should do is, every erro that occurs should fail the payment in the payment and wait for all collectors to resolve.